nomt/src/lib.rs

```diff
@@ -259,17 +261,34 @@ impl<T: HashAlgorithm> Nomt<T> {
         self.store.load_value(path)
     }
 
-    /// Creates a new [`Session`] object, that serves a purpose of capturing the reads and writes
-    /// performed by the application, updating the trie and creating a [`Witness`], allowing to
-    /// re-execute the same operations without having access to the full trie.
-    ///
-    /// Only a single session may be created at a time. Creating a new session without dropping or
-    /// committing an existing open session will lead to a panic.
+    /// Creates a new [`Session`]. Only a single session may be active at a time.
     pub fn begin_session(&self) -> Session {
-        self.begin_session_inner(/* allow_rollback */ true)
+        // UNWRAP: empty ancestors are always valid.
+        self.begin_session_inner(/* allow_rollback */ true, None)
+            .unwrap()
+    }
+
+    /// Creates a new [`Session`] object on top of a series of in-memory overlays.
+    ///
+    /// This will fail if the first overlay was not created on top of the previous overlays.
+    /// Providing an empty iterator will give the same result as `begin_session`.
+    ///
+    /// It is the user's responsibility to ensure that all uncommitted overlays are provided.
+    /// Failing to provide all uncommitted overlays will lead to incomplete proofs or corruption.
```

Comment:
Likewise here. This also seems super error-prone.

I don't know if there's a legitimate reason to ever omit these, and it would be better to fail with a nice error instead.